R/conda.R

```diff
@@ -182,14 +182,13 @@ conda_install <- function(envname = NULL,
 
   if (pip) {
     # use pip package manager
-    condaenv_bin <- function(bin) path.expand(file.path(dirname(conda), bin))
-    cmd <- sprintf("%s%s %s && pip install --upgrade %s %s%s",
-                   ifelse(is_windows(), "", ifelse(is_osx(), "source ", "/bin/bash -c \"source ")),
-                   shQuote(path.expand(condaenv_bin("activate"))),
+    cmd <- sprintf("%s %s && pip install --upgrade %s %s%s",
+                   ifelse(is_windows(), paste(conda, "activate"),
+                          sprintf("bash -c \"eval $(%s shell.bash hook) && conda activate", conda)),
```

Comment:
For my own context, where is the `eval $(%s shell.bash hook)` coming from?

Reply:
This is because in recent versions of Conda the `conda` command is no longer an executable file but a function in your shell. You can check this by running `which conda`. That's okay as long as you are using your shell interactively, but functions are not exported to non-interactive subshells. The `conda shell.bash hook` command makes the function available in the subshell. I took it from this post: https://github.com/conda/conda/issues/7980#issuecomment-492784093.